### PR TITLE
add CLI commands for ocr2vrf deployment, configuration and request

### PR DESCRIPTION
**Pull request by jinhoonbang**
Test commands:

```
go run . coordinator-deploy --dkg-address 0x3e54dCc49F16411A3aaa4cDbC41A25bCa9763Cee --link-eth-feed-address 0xFABe80711F3ea886C3AC102c81ffC9825E16162E --key-id 6f3b82406688b8ddb944c6f2e6d808f014c8fa8d568d639c25019568c715fbf0 --beacon-period-blocks 5

Executing contract deployment, TX: https://rinkeby.etherscan.io/tx/0x879d13012495cd8219b31412e3c021d685cd47aa78c8a547f147926543f6bb96
Contract Address: 0x5f0776736Ae1da43936b60546881E41b9100D0bD


go run . coordinator-set-config --coordinator-address "0x5f0776736Ae1da43936b60546881E41b9100D0bD" \
--key-id "6f3b82406688b8ddb944c6f2e6d808f014c8fa8d568d639c25019568c715fbf0" \
 --onchain-pub-keys "0a8cbea12a06869d3ec432ab9682dab6c761d591,f4f9db7bb1d16b7cdfb18ec68994c26964f59853,22fb3f90c539457f00d8484438869135e604a655,33cbcedccb11c9773ad78e214ba342e979255ab3,6ffaa96256fbc1012325cca88c79f725c33eed80"  \
 --offchain-pub-keys "6c5ca6f74b532222ac927dd3de235d46a943e372c0563393a33b01dcfd3f371c,855114d25c2ef5e85fffe4f20a365672d8f2dba3b2ec82333f494168a2039c04,0266e835634db00977cbc1caa4db10e1676c1a4c0fcbc6ba7f09300f0d183182,980cd91f7a73f20f4b0d51d00cd4e00373dc2beafbb299ca3c609757ab98c830,eb6d36e2af8922085ff510bbe1eb8932a0e3295ca9f047fef25d90e69c52948f" \
 --config-pub-keys "1706390f0e3a7e666837178ba85657ac1f7c0ccee474db1b11b92894bb471f0e,21fb71681ad745dc82203b794f13989dc33f955872d526477ff776f024e3ca6e,c26fd919fcb37517b24e535d223ed474900224cd6bbf43b55ba8908e03a60c28,1518fae281ba06cd499af3fd7cc0b4853408e9dfe088d30de6479384706ae832,17572614924936d0bbc35c679f246972dad79500c2d2461049b5cc9a1318d57d" \
 --peer-ids "12D3KooWDc6Kr2dEBhKY2k3nhPWiFvTEe2S1p9xTE2TKt4Murqoh,12D3KooWKCnCgrKcwC2J5wWjbn5T53PhdkkoWENSJ9Te7TKx66oJ,12D3KooWR9aogYHxk5z8ckbLLV4nBozwztxqfJpPXfq3mJr2ydRG,12D3KooWDiTDcUegV7wk113G3fGjiabYuoTCo1WrooSegA42cUoT,12D3KooWNdhprXkTrfSp5MPqsbpFzp6Ag9JSxsXiCACDBgdTBLef" \
 --transmitters "74103Cf8b436465870b26aa9Fa2F62AD62b22E35,38A6Cb196f805cC3041F6645a5A6CEC27B64430D,47d7095CFEBF8285BdAa421Bc8268D0DB87D933C,a8842BE973800fF61D80d2d53fa62C3a685380eB,3750e31321aEE8c024751877070E8d5F704cE987" \
 --schedule "1,1,1,1,1"

confDelays: map[1:{} 2:{} 3:{} 4:{} 5:{} 6:{} 7:{} 8:{}]
vrf beacon coordinator offchain config: 6f3b82406688b8ddb944c6f2e6d808f014c8fa8d568d639c25019568c715fbf0
vrf beacon coordinator onchain config: 00000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000005000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000070000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000003
Executing TX https://rinkeby.etherscan.io/tx/0x3fc7f60c462911ce6fb3ee7d330e39e547188a83cf74d77e6408260df64159fd []
TX 0x3fc7f60c462911ce6fb3ee7d330e39e547188a83cf74d77e6408260df64159fd mined.
Block Number: 11049002
Gas Used:  694696

go run . coordinator-request-randomness --coordinator-address 0x5f0776736Ae1da43936b60546881E41b9100D0bD --sub-id 0 --conf-delay 2

Executing TX https://rinkeby.etherscan.io/tx/0x8ea2abcfd95a04738c058194c4db30f6bf3d15954873e584b555decdf4041d52 []
TX 0x8ea2abcfd95a04738c058194c4db30f6bf3d15954873e584b555decdf4041d52 mined.
Block Number: 11049007
Gas Used:  74786

go run . coordinator-get-randomness --coordinator-address 0x5f0776736Ae1da43936b60546881E41b9100D0bD --request-id [requestID]
panic: execution reverted

go run . consumer-deploy --coordinator-address 0x5f0776736Ae1da43936b60546881E41b9100D0bD --beacon-period-blocks 5

Executing contract deployment, TX: https://rinkeby.etherscan.io/tx/0xb56feb373f4be23011f257b540516566f93c17cc6c525bf9911b6af4e99e30dc
Contract Address: 0xf051EF99C1801c6c6C67cFea680FCd605A51fed7

go run . consumer-request-randomness --consumer-address 0xf051EF99C1801c6c6C67cFea680FCd605A51fed7 --sub-id 1 --conf-delay 2

Executing TX https://rinkeby.etherscan.io/tx/0x9b1195523f4ffb60548991f045982a2e4e289b313a1b77182e7b88a03666a92b []
TX 0x9b1195523f4ffb60548991f045982a2e4e289b313a1b77182e7b88a03666a92b mined.
Block Number: 11052843
Gas Used:  110748
nextBeaconOutputHeight:  11052845
requestID:  2

go run . consumer-get-randomness --consumer-address 0xf051EF99C1801c6c6C67cFea680FCd605A51fed7 --request-id 2

panic: execution reverted

goroutine 1 [running]:
github.com/smartcontractkit/chainlink/core/scripts/common.PanicErr(...)
    /Users/jinbang/development/chainlink/core/scripts/common/helpers.go:104
main.getRandomnessFromConsumer({0x14000dc0230?, 0x140001520a0?, 0xa?}, {0x16ae93647?, 0x1400289fbe8?}, 0x14000dbe0e0)
    /Users/jinbang/development/chainlink/core/scripts/ocr2vrf/util.go:283 +0x114
main.main()
    /Users/jinbang/development/chainlink/core/scripts/ocr2vrf/main.go:245 +0x10b8
exit status 2
```

### CODE REVIEW

**Review by vreff — COMMENTED** [review]
LGTM pending Makram's comments and updating the go.mod. I will say it seems like a weird user experience for the execution to revert on getRandomness if the request is unfulfilled... but that has nothing to do with these scripts.

**Comment by jinhoonbang** [comment]
> LGTM pending Makram's comments and updating the go.mod. I will say it seems like a weird user experience for the execution to revert on getRandomness if the request is unfulfilled... but that has nothing to do with these scripts.

+1 on this. It doesn't seem conventional to have a mutating get* call.

File: core/scripts/ocr2vrf/main.go [hunk 1]
```diff
@@ -35,6 +36,11 @@ type dkgSetConfigArgs struct {
 	dkgSigningPubKeys    string
 	keyID                string
 }
```
[the remark this answers is not in the document]
Comment (jinhoonbang) [reply core/scripts/ocr2vrf/main.go:38]: hmm interesting. somehow `goimports -w -local github.com/smartcontractkit/chainlink ./` didn't fix this for me. Updated manually for now

**Comment by makramkd** [comment]
Good find, I think the method is just misnamed. It should be called something like `fulfillRandomness` instead.

File: core/scripts/ocr2vrf/util.go [hunk 2]
```diff
@@ -248,6 +253,37 @@ func getRandomness(e helpers.Environment, coordinatorAddress string, requestID *
...
+	periodBlocks, err := consumer.IBeaconPeriodBlocks(nil)
+	helpers.PanicErr(err)
+
+	blockNumber := receipt.BlockNumber
+	periodOffset := new(big.Int).Mod(blockNumber, periodBlocks)
+	nextBeaconOutputHeight := new(big.Int).Sub(new(big.Int).Add(blockNumber, periodBlocks), periodOffset)
+
+	fmt.Println("nextBeaconOutputHeight: ", nextBeaconOutputHeight)
+
+	requestID, err := consumer.SRequestsIDs(nil, nextBeaconOutputHeight, confDelay)
```
Comment (vreff) [inline core/scripts/ocr2vrf/util.go:272]: So this returns the most recent requestID for that beacon height/confDelay combo, right?

Comment (jinhoonbang) [reply core/scripts/ocr2vrf/util.go:272]: yeah that's right. seems like the consumer contract implementation overrides the requestID for the same height/confDelay

**Review by makramkd — APPROVED** [review]

**Pull request merged**